### ISSUE TITLE
Ledges for Tux to traverse

### ISSUE THREAD

**Issue by SuperTux20** [SuperTux/supertux#1335]
## This ledge mechanic could be useful in a future mountain world, and possibly in other places too. (Caves?)

Tux moves slowly on ledges, and he can switch between hanging from them to shimmying across them at any time by pressing the down and up arrow keys.
Ledges would also be unisolid, jump up from below and Tux would hang from them.
Tux can also jump off of the ledges when hanging, and jump up to avoid/stomp a flying enemy when shimmying.
Walking enemies treat ledges like empty space, they either fall off or turn around.

![image](https://user-images.githubusercontent.com/57378284/73412218-c97e3d80-42bc-11ea-9c03-8f0ddc7ce630.png)

###### *This is my first time submitting a "feature" issue. Am I doing it right?*

**Comment by busybody42** [comment]
https://forum.freegamedev.net/viewtopic.php?f=67&t=12847 Did you pay attention to DevonST's advice about posting it on github?

**Comment by SuperTux20** [comment]
> Did you pay attention to DevonST's advice about posting it on github?

Yeah, he didn't leave me any tips or anything though. He just said "so it doesn't get lost"

**Comment by Ordoviz** [comment]
You did everything right. Did the issue template make you insecure about posting a feature request? Maybe we should add a note to the top:

> If your issue is not a bug report, you can safely remove the following lines.

**Comment by SuperTux20** [comment]
> Did the issue template make you insecure about posting a feature request?
Yeah, a little bit. Then again, I get insecure about a lot of things.

**Comment by HybridDog** [comment]
I don't like ledges in NSMB. I think before implementing this or other features from Mario Bros, the current buggy features should be fixed, e.g. the ladder which accelerates Tux to itself when pressing up and which drops Tux when he climbs too high. I personally also prefer unique features, such as sliding. 
![original](https://www.mariowiki.com/images/7/7d/6-4_NSMB.png)

**Comment by SuperTux20** [comment]
> The current buggy features should be fixed, e.g. the ladder which accelerates Tux to itself when pressing up and which drops Tux when he climbs too high.

Oof, yeah you're probably right. I hate that bug.

**Comment by busybody42** [comment]
who says that they have to act EXACTLY like the ledges from NSMB DS?

**Comment by HybridDog** [comment]
> who says that they have to act EXACTLY like the ledges from NSMB DS?

Nobody says that, but the pictures imply that they're the exact same ledges.
